doc/how_to_introduce_breaking_changes.md

```diff
@@ -72,7 +101,7 @@ You can then pre-announce breaking changes for different command groups or comma
 from azure.cli.core.breaking_change import register_required_flag_breaking_change, register_default_value_breaking_change, register_other_breaking_change
 
 register_required_flag_breaking_change('bar foo', '--name')
-register_default_value_breaking_change('bar foo baz', '--foobar', 'A', 'B')
+register_default_value_breaking_change('bar foo baz', '--foobar', 'A', 'B', target_version='May 2025')
```

Comment:
May I ask if our target version is currently using the date style?

Reply:
I support using plain text in this PR, so that we could support date style. This will be beneficial for an upcoming breaking change in an extension, which could occur in the coming months.

I don't believe we need to ensure consistency in the target version style, as version numbers are unpredictable for both developers and customers. While they know the version, they don't know when it will be released. Therefore, it is more flexible to support both date and version formats.

Reply:
Yes, I agree with your~ But what I mean is that since we all use date styles, is it necessary to have a unified style to avoid some people writing the wrong date or inconsistent date styles? Such as some people write `May 2025` while others write `2025-05`、`05/2025`.....

Reply:
It is not easy to constraint. I think it's ok to keep the flexibility of a field with any str.
Usually, developers would follow the example in doc. And I will also add instructions about date format in doc.

Reply:
OK, LGTM